lib/connect.js

```diff
@@ -633,7 +658,34 @@ function validateBandwidthProfile(bandwidthProfile) {
     }
     return typeof video[prop] === typeOrValues ? null
       : Error(`options.bandwidthProfile.video.${prop}`, typeOrValues);
+  }, validateRenderDimensions(video.renderDimensions));
+}
+
+function validateDimensions(prop, dimensions) {
+  if (typeof dimensions === 'undefined') {
+    return null;
+  }
+  if (dimensions === null || !isNonArrayObject(dimensions)) {
+    return E.INVALID_TYPE(`options.bandwidthProfile.video.renderDimensions.${prop}`, 'object');
+  }
+  return ['height', 'width'].reduce((error, dimension) => {
+    if (error || !(dimension in dimensions)) {
+      return error;
+    }
+    return typeof dimensions[dimension] === 'number' ? null
+      : E.INVALID_TYPE(`options.bandwidthProfile.video.renderDimensions.${prop}.${dimension}`, 'number');
   }, null);
 }
 
+function validateRenderDimensions(renderDimensions) {
+  if (typeof renderDimensions === 'undefined') {
+    return null;
+  }
+  if (renderDimensions === null || !isNonArrayObject(renderDimensions)) {
```

Comment:
I see that pretty much all usages of `isNonArrayObject` are preceded by checking for null. Consider moving it inside the function.

Reply:
@makarandp0 We use `isNonArrayObject` in other places where `null` is a valid parameter.